app/controllers/two_factor_authentication/otp_verification_controller.rb

```diff
@@ -21,6 +21,7 @@ def create
       result = otp_verification_form.submit
       post_analytics(result)
       if result.success?
+        handle_valid_confirmation_otp if UserSessionContext.confirmation_context?(context)
         handle_valid_otp(next_url: nil, auth_method: params[:otp_delivery_preference])
```

Comment:
Would it make sense to pull `handle_valid_otp` into this class? And then maybe consolidate this to a single "handle_valid_*" call?

Reply:
session context really only exists for phone OTPs, so I'm leaning towards making the `handle_valid_confirmation`/`handle_valid_authentication` calls explicit, and the phone controller can have the conditional checks with `UserSessionContext.confirmation_context?`, etc.

I'm ultimately hoping this controller will be something like:

```ruby
if UserSessionContext.confirmation_context?(context)
  handle_valid_confirmation_otp
  handle_valid_verification_for_confirmation
else
  handle_valid_verification_for_authentication
end
```

and the remaining auth controllers can call `handle_valid_verification_for_confirmation` / `handle_valid_verification_for_authentication` without using session context since the controller action should be enough to know the difference.